.travis.yml

```diff
@@ -15,6 +15,6 @@ script:
   - flake8 *.py
   - pytest run_tests.py
 after_success:
-  - pip install -q coverage==4.4 coveralls pytest-cov
+  - pip install -q 'coverage==4.4' 'coveralls<2' 'pytest-cov<3'
```

Comment:
This probably shouldn't be done in CI

Reply:
Pinning helps the developer/CI environment evolve in a controlled manner, so that updates can be done separately from other PRs and not on an emergency basis. In particular:

The `coveralls` package actually [recommends pinning](https://pypi.org/project/coveralls/) in this case. This Travis config includes a python2.7 config on xenial -- and [xenial includes pip 8](https://packages.ubuntu.com/xenial/python-pip) by default. Support for data-requires-python was added in [pip 9.0.0](https://pip.pypa.io/en/stable/news/).

The `pytest-cov` plugin [adds command line options](https://github.com/pytest-dev/pytest-cov/blob/da6ad9454910ff0a7c792d3d906660f77806f6b5/src/pytest_cov/plugin.py#L64); a new major version could change those and break the `pytest` command.  (`isort` 5 also made major changes to the options.)

However, none of this is actually exercised right now because the flake8 checks are currently failing.

Reply:
This is true but seems to be unrelated to the dist metadata change. I'd say that env pins should use [constraints lock files](https://pip.pypa.io/en/stable/reference/pip_install/#install-constraint) and probably be managed by [pip-tools](https://github.com/jazzband/pip-tools) that is also compatible with [Dependabot that is now built into GitHub itself](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/).
So this is a wider issue and would deserve to be in the separate (bigger?) PR.

Reply:
The relation is in the common problem (risk of breakage due to missing upper bound); when I see an issue I have a habit of fixing it everywhere in a project.

Great point about Dependabot (or the many great alternatives such as pyup.io) to automatically update pins. I do agree an exact pin does reduce the risk of breakage further and a bot helps make the updating easier. But this has to be something a maintainer wants. It needs to be set up, and also requires a commitment to review and merge the bot's PRs in a timely manner.

Reply:
Descoping .travis.yml from this PR. Will do the pip-tools way under separate PR.